src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -96,4 +96,7 @@
 
     @Query("select distinct team.students from Team team where team.exercise.course.id = :#{#courseId} and team.shortName = :#{#teamShortName}")
     Set<User> findAllInTeam(@Param("courseId") Long courseId, @Param("teamShortName") String teamShortName);
+
+    @Query("SELECT u FROM User u JOIN FETCH u.achievements a WHERE a.id = :#{#achievementId}")
+    Set<User> findAllWithEagerAchievements(@Param("achievementId") Long achievementId);
```

Comment:
Should be `findAllWithEagerAchivement` or not? 
I think this would be more understandable, when you do it the other way around. 
So in the AchievementRepository and make `findByIdWithEagerUsers(achievementId)`

Reply:
yeah, would also work 😄

Reply:
sounds goo! :)